jujugui/static/gui/src/app/components/isv-profile/plans-usage/plans-usage.js

```diff
@@ -0,0 +1,170 @@
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*
+The ISV profile component renders a venders profile with stats based on there
+charms and bundles. You can only see this page is you have access.
+*/
+
+'use strict';
+
+YUI.add('plans-usage', function() {
+
+  juju.components.PlansUsage = React.createClass({
+    propTypes: {
+      d3: React.PropTypes.object.isRequired,
+      dataset: React.PropTypes.array.isRequired
+    },
+
+    componentDidMount: function() {
+      this.renderChart();
+      this.setState({d3Rendered: true});
+    },
+
+    shouldComponentUpdate: function shouldComponentUpdate() {
+      return !this.state.d3Rendered;
+    },
+
+    /**
+      Get the initial state.
+
+      @method getInitialState
+      @returns {String} The intial state.
+    */
+    getInitialState: function() {
+      return {
+        d3Rendered: false
+      };
+    },
+
+    /**
+      Plot a line on the graph
+
+      @method plotLine
+      @param {Array} The line data
+    */
+    plotLine: function(data) {
+      const d3 = this.props.d3;
+      const lineGen = d3.svg.line()
+        .x(function(d) {
+          return this.xScale(d.date);
+        }.bind(this))
+        .y(function(d) {
+          return this.yScale(d.value);
+        }.bind(this));
+
+      this.svg.append('path')
+        .attr('d', lineGen(data))
+        .attr('stroke', 'teal')
+        .classed('line', true);
+    },
+
+    /**
+      Render the base chart, including the axises.
+
+      @method renderChart
+    */
+    renderChart: function() {
+      const dataset = this.props.dataset;
+      const d3 = this.props.d3;
+      const w = 700;
+      const h = 500;
+      const ticks = dataset[0].length;
+      this.margins = {
+        top: 20,
+        right: 20,
+        bottom: 20,
+        left: 50
+      };
+
+      const flattenedDataset = dataset.reduce(function(a, b) {
+        return a.concat(b);
+      }, []);
+
+      const x_domain = d3.extent(flattenedDataset, function(d) {
+        return d.date;
+      });
+
+      const y_domain = d3.extent(flattenedDataset, function(d) {
+        return d.value;
+      });
+
+      // Display date format (05/16)
+      const date_format = d3.time.format('%m/%d');
+
+      // Create an SVG container
+      const vis = d3.select('.d3-chart-wrapper');
+      this.svg = vis.append('svg')
```

Comment:
I tried to use state to store these variables but as setState is not instant. I cannot guarantee that they will be set throughout the class. Is there a way of having class global variables?